lib/addons/src/types.ts

```diff
@@ -136,16 +142,16 @@ export interface StoryApi<StoryFnReturnType = unknown> {
 }
 
 export type DecoratorFunction<StoryFnReturnType = unknown> = (
-  fn: StoryFn<StoryFnReturnType>,
+  fn: PartialStoryFn<StoryFnReturnType>,
   c: StoryContext
-) => ReturnType<StoryFn<StoryFnReturnType>>;
+) => ReturnType<LegacyStoryFn<StoryFnReturnType>>;
```

Comment:
Why is this changed to `LegacyStoryFn`?

Reply:
Decorators deal in `LegacyStoryFn`s always (they never expect to pass args as the first argument, or receive it).

In the store, we transform `original` (an `ArgsStoryFn`) into `finalStoryFn` (a `LegacyStoryFn`) before decorating it. When we actually render the decorated story, we pass `context` in as first argument.

Reply:
I see ✅